master/buildbot/config.py

```diff
@@ -900,7 +901,7 @@ def check_lock(lock):
                 lock_dict[lock.name] = lock
 
         for b in self.builders:
-            if b.locks:
+            if b.locks and not IRenderable.providedBy(b.locks):
```

Comment:
This check is needed because renderable locks are not known until builder start and we can't do any checks at config time for these locks.